app/webpacker/components/Panel/Board/BoardEditorPage/BoardEditor.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import {
+  Button, Form, Header, Modal, Table,
+} from 'semantic-ui-react';
+import { useConfirm } from '../../../../lib/providers/ConfirmProvider';
+import useSaveAction from '../../../../lib/hooks/useSaveAction';
+import { apiV0Urls } from '../../../../lib/requests/routes.js.erb';
+import { groupTypes } from '../../../../lib/wca-data.js.erb';
+import Loading from '../../../Requests/Loading';
+import SEARCH_MODELS from '../../../SearchWidget/SearchModel';
+import WcaSearch from '../../../SearchWidget/WcaSearch';
+import useInputState from '../../../../lib/hooks/useInputState';
+
+export default function BoardEditor({ boardRoles, sync }) {
+  const [openModal, setOpenModal] = useState(false);
+  const [newBoardRole, setNewBoardRole] = useInputState(null);
+  const { save, saving } = useSaveAction();
+  const confirm = useConfirm();
+
+  const endRole = (boardRole) => {
+    confirm().then(() => {
+      save(apiV0Urls.userRoles.delete(boardRole.id), {}, sync, { method: 'DELETE' });
+    });
+  };
+
+  if (saving) return <Loading />;
+
+  return (
+    <>
+      <Header>Board Editor</Header>
+      <Button onClick={() => setOpenModal(true)}>Add Role</Button>
+      <Table celled>
+        <Table.Header>
+          <Table.Row>
+            <Table.HeaderCell>Name</Table.HeaderCell>
+            <Table.HeaderCell>Actions</Table.HeaderCell>
+          </Table.Row>
+        </Table.Header>
+        <Table.Body>
+          {boardRoles.map((boardRole) => (
+            <Table.Row key={boardRole.id}>
+              <Table.Cell>{boardRole.user.name}</Table.Cell>
+              <Table.Cell>
+                <Button onClick={() => endRole(boardRole)}>End Role</Button>
+              </Table.Cell>
+            </Table.Row>
+          ))}
+        </Table.Body>
+      </Table>
+      {openModal && (
+      <Modal
+        size="fullscreen"
+        open={openModal}
+        onClose={() => {
+          setOpenModal(false);
+          newBoardRole(null);
```

Comment:
Shouldn't this be `setNewBoardRole`? In that case, you can extract it into its own function and use that new function here and also in the `onSubmit` success fn below.

Reply:
Oh yes, done. Thanks.